### PR TITLE
readme, contributing: rewrite and reorganize contribution resources

### PR DESCRIPTION
**Pull request by lilin90**
### What is changed, added or deleted? 

- Rewrite `README.md` to make it more like a docs readme (replace simply intro to TiDB)
- `CONTRIBUTING.md` updates
    - Add more resources
    - Convert the list of 26 markdownlint rules to a table (more friendly for users to read only the Chinese explanation)
    - Reorganize the items of markdownlint rules to make it more logical in content (such as keeping titles rules together)
- Update inappropriate and obsolete links in `README.md` and `CONTRIBUTING.md`

### What is the related PR or file link(s)? 

N/A

### Which version does your change affect? 

N/A

### CODE REVIEW

**Comment by lilin90** [comment]
@yikeke PTAL~

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,145 +1,91 @@
...
+| 4 | [MD019 - Multiple spaces after hash on atx style heading](https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#md019---multiple-spaces-after-hash-on-atx-style-heading) | 标题的引导符号“#”后只能空**一格**后再接标题内容，不能有多个空格。 |
+| 5 | [MD023 - Headings must start at the beginning of the line](https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#md023---headings-must-start-at-the-beginning-of-the-line) | 标题必须出现在一行行首，即标题的 `#` 号前不能有任何空格。 |
+| 6 | [MD026 - Trailing punctuation in heading](https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#md026---trailing-punctuation-in-heading) | 标题末尾仅能出现中英文问号、反引号、中英文单双引号等符号。其余如**冒号**、逗号、句号、感叹号等符号均不能在标题末尾使用。 |
+| 7 | [MD022 - Headings should be surrounded by blank lines](https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#md022---headings-should-be-surrounded-by-blank-lines) | 标题上下均须空一行。 |
+| 8 | [MD024 - Multiple headings with the same content](https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#md024---multiple-headings-with-the-same-content) | 文档中不能连续出现内容重复的标题，如一级标题为 `# TiDB 架构`，紧接着的二级标题就不能是 `# TiDB 架构`。如果不是连续的标题，则标题内容可重复。 |
```
Comment (ran-huang) [inline CONTRIBUTING.md]: Is this a typo? 😀
```suggestion
| 8 | [MD024 - Multiple headings with the same content](https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#md024---multiple-headings-with-the-same-content) | 文档中不能连续出现内容重复的标题，如一级标题为 `# TiDB 架构`，紧接着的二级标题就不能是 `## TiDB 架构`。如果不是连续的标题，则标题内容可重复。 |
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,145 +1,91 @@
...
+| 17 | [MD030 - Spaces after list markers](https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#md030---spaces-after-list-markers) | 使用列表时，每一列表项的标识符（`+`、`-`、`*` 或数字）后只能**空一格**，后接列表内容。|
+| 18 | [MD032 - Lists should be surrounded by blank lines](https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#md032---lists-should-be-surrounded-by-blank-lines) | 列表（包括有序和无序列表）前后必须各空一行。 |
+| 19 | [MD031 - Fenced code blocks should be surrounded by blank lines](https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#md031---fenced-code-blocks-should-be-surrounded-by-blank-lines) | 代码块前后必须各空一行。 |
+| 20 | [MD034 - Bare URL used](https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#md034---bare-url-used) | 文档中禁止出现裸露的 URL。如果希望用户能直接点击并打开该 URL，则使用一对尖括号 (`<URL>`) 包裹该 URL。如果由于特殊情况必须使用裸露的 URL，不需要用户通过点击打开，则使用一对反引号 (``` `URL` ```) 包裹该 URL。 |
+| 21 | [MD037 - Spaces inside emphasis markers](https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#md037---spaces-inside-emphasis-markers) | 使用加粗、斜体等强调效果时，在强调标识符内禁止出现多余的空格。如不能出现 ``` `** 加粗文本 **` ```。 |
```
Comment (ran-huang) [inline CONTRIBUTING.md]: I don't understand the usage of three backticks. Is this a special Markdown syntax?
```suggestion
| 21 | [MD037 - Spaces inside emphasis markers](https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#md037---spaces-inside-emphasis-markers) | 使用加粗、斜体等强调效果时，在强调标识符内禁止出现多余的空格。如不能出现 `** 加粗文本 **`。 |
```

File: CONTRIBUTING.md (hunk 1 above)
Comment (lilin90) [reply CONTRIBUTING.md]: @ran-huang Yes. Good catch! Would you please help fix this typo in all versions? (including release-2.0 & release-1.0)

File: CONTRIBUTING.md (hunk 2 above)
Comment (lilin90) [reply CONTRIBUTING.md]: @ran-huang np~ Just to find a way to display what we want to show.

File: CONTRIBUTING.md (hunk 1 above)
Comment (ran-huang) [reply CONTRIBUTING.md]: Sure, I'll fix this later~

**Pull request merged**